### PR TITLE
Adding Dockerfile for AMD based on Centos9

### PR DESCRIPTION
**Pull request by dibryant**
FIxes for https://issues.redhat.com/browse/RHOAIENG-4368
## Description

Docker file for ROCM based on: https://github.com/ROCm/ROCm-docker/blob/master/dev/Dockerfile-centos-7-complete

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: amd/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,93 @@
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+
+ARG ROCM_VERSION=5.3
+ARG AMDGPU_VERSION=5.3
+# Base
+RUN yum -y install git java-1.8.0-openjdk python; yum clean all
+
+# Enable epel-release repositories
```
Comment (atheo89) [inline amd/Dockerfile:9]: Could you add as comment the link of the source code that you fetch these dockerfile instructions please?

Comment (dibryant) [reply amd/Dockerfile:9]: > Could you add as comment the link of the source code that you fetch these dockerfile instructions please?
[
](https://github.com/ROCm/ROCm-docker/blob/master/dev/Dockerfile-centos-7-complete)

Comment (atheo89) [reply amd/Dockerfile:9]: Thx for the link! Do you mind to comment the link inside on Dockerfile? like:
```suggestion
# Install ROCm AMD from:
# https://github.com/ROCm/ROCm-docker/blob/master/dev/Dockerfile-centos-7-complete

# Enable epel-release repositories

File: amd/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,117 @@
+ARG BASE_IMAGE
```
Comment (atheo89) [inline amd/Dockerfile:1]: Could you please create a folder and then add the Dockefile file in it?  Should be `amd/ubi9-python-3.9/Dockerfile`
  
We this we follow the same folder pattern that we do for all the rest. For your ref,  check out  here: https://github.com/opendatahub-io/notebooks/tree/main/cuda

Comment (dibryant) [reply amd/Dockerfile]: Yes instead it would be c9s-python-3.9 we are holding off on ubi for right now

File: amd/c9s-python-3.9/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      authoritative-source-url="https://github.com/opendatahub-io/notebooks" \
+      io.openshift.build.commit.ref="main" \
+      io.openshift.build.source-location="https://github.com/opendatahub-io/notebooks/tree/main/amd/python-3.9" \
+      io.openshift.build.image="quay.io/opendatahub/workbench-images:rocm-python-3.9"
+
+USER 0
+WORKDIR /opt/app-root/bin
+
+ARG ROCM_VERSION=5.3
+ARG AMDGPU_VERSION=5.3
```
Comment (harshad16) [inline amd/c9s-python-3.9/Dockerfile:17]:
```suggestion
ARG ROCM_VERSION=6.1
ARG AMDGPU_VERSION=6.1
```

File: amd/c9s-python-3.9/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+# Base
+RUN yum -y install git java-1.8.0-openjdk python; yum clean all
+# Install ROCm AMD from:
+# https://github.com/ROCm/ROCm-docker/blob/master/dev/Dockerfile-centos-7-complete
+# Enable epel-release repositories
+# RUN yum --enablerepo=extras install -y epel-release
+RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && \
+    dnf clean all && rm -rf /var/cache/yum
```
Comment (harshad16) [inline amd/c9s-python-3.9/Dockerfile:25]:
```suggestion
# Enable epel-release repositories
RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
    dnf config-manager --set-enabled crb && \
    dnf clean all && rm -rf /var/cache/yum
```

File: amd/c9s-python-3.9/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+# elfutils-libelf-devel \    needed by dkms \
+# expect \
+# python3-dev \
+# libcxx-devel \
+# numactl-devel \
+# pciutils-devel \
+# pth \
+# qemu-kvm \
+# re2c \
+# subversion \
```
Comment (harshad16) [inline amd/c9s-python-3.9/Dockerfile:81]:
```suggestion
# Install required base build and packaging commands for ROCm
RUN dnf -y install \
    bc \
    bridge-utils \
    glibc.i686 \
    numactl-libs \
    libssh \
    libunwind-devel \
    libunwind \
    pciutils \
    pciutils-libs \
    re2c \
    doxygen \
    elfutils-libelf-devel \
    expect \
    numactl-devel \
    pciutils-devel \
    qemu-kvm \
    subversion \
    dkms \
    dpkg \
    dpkg-dev \
    dpkg-perl

# These binaries are not available.
# devscripts \
# pth \
# libcxx-devel \
```

File: amd/c9s-python-3.9/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+# RUN yum install -y centos-release-scl
+
+# Install the devtoolset-7 collection:
+# RUN yum install -y devtoolset-7
+# RUN yum install -y devtoolset-7-libatomic-devel devtoolset-7-elfutils-libelf-devel
+
+# Install the ROCm rpms
+RUN yum clean all
+RUN echo -e "[ROCm]\nname=ROCm\nbaseurl=https://repo.radeon.com/rocm/yum/$ROCM_VERSION/main\nenabled=1\ngpgcheck=0" >> /etc/yum.repos.d/rocm.repo
+RUN echo -e "[amdgpu]\nname=amdgpu\nbaseurl=https://repo.radeon.com/amdgpu/$AMDGPU_VERSION/rhel/7.9/main/x86_64\nenabled=1\ngpgcheck=0" >> /etc/yum.repos.d/amdgpu.repo
```
Comment (harshad16) [inline amd/c9s-python-3.9/Dockerfile:101]:
```suggestion
# Install the ROCm rpms
RUN dnf clean all
RUN echo "[ROCm]" > /etc/yum.repos.d/rocm.repo && \
    echo "name=ROCm" >> /etc/yum.repos.d/rocm.repo && \
    echo "baseurl=https://repo.radeon.com/rocm/el9/$ROCM_VERSION/main" >> /etc/yum.repos.d/rocm.repo && \
    echo "enabled=1" >> /etc/yum.repos.d/rocm.repo && \
    echo "gpgcheck=0" >> /etc/yum.repos.d/rocm.repo

RUN echo "[amdgpu]" > /etc/yum.repos.d/amdgpu.repo && \
    echo "name=amdgpu" >> /etc/yum.repos.d/amdgpu.repo && \
    echo "baseurl=https://repo.radeon.com/amdgpu/$AMDGPU_VERSION/el/9.2/main/x86_64" >> /etc/yum.repos.d/amdgpu.repo && \
    echo "enabled=1" >> /etc/yum.repos.d/amdgpu.repo && \
    echo "gpgcheck=0" >> /etc/yum.repos.d/amdgpu.repo
```

File: amd/c9s-python-3.9/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+# Set ENV to enable devtoolset7 by default 
+ENV PATH=/opt/rh/devtoolset-7/root/usr/bin:/opt/rocm/hcc/bin:/opt/rocm/hip/bin:/opt/rocm/bin:/opt/rocm/hcc/bin:${PATH:+:${PATH}}
+ENV MANPATH=/opt/rh/devtoolset-7/root/usr/share/man:${MANPATH}
+ENV INFOPATH=/opt/rh/devtoolset-7/root/usr/share/info${INFOPATH:+:${INFOPATH}}
+ENV PCP_DIR=/opt/rh/devtoolset-7/root
+ENV PERL5LIB=/opt/rh/devtoolset-7/root//usr/lib64/perl5/vendor_perl:/opt/rh/devtoolset-7/root/usr/lib/perl5:/opt/rh/devtoolset-7/root//usr/share/perl5/
+ENV LD_LIBRARY_PATH=/opt/rocm/lib:/usr/local/lib:/opt/rh/devtoolset-7/root$rpmlibdir$rpmlibdir32${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
+ENV PYTHONPATH=/opt/rh/devtoolset-7/root/usr/lib64/python$pythonvers/site-packages:/opt/rh/devtoolset-7/root/usr/lib/python$pythonvers/
+ENV LDFLAGS="-Wl,-rpath=/opt/rh/devtoolset-7/root/usr/lib64 -Wl,-rpath=/opt/rh/devtoolset-7/root/usr/lib"
```
Comment (harshad16) [inline amd/c9s-python-3.9/Dockerfile:114]:
```suggestion
# Set ENV to enable devtoolset7 by default 
# ENV PATH=/opt/rh/devtoolset-7/root/usr/bin:/opt/rocm/hcc/bin:/opt/rocm/hip/bin:/opt/rocm/bin:/opt/rocm/hcc/bin:${PATH:+:${PATH}}
# ENV MANPATH=/opt/rh/devtoolset-7/root/usr/share/man:${MANPATH}
# ENV INFOPATH=/opt/rh/devtoolset-7/root/usr/share/info${INFOPATH:+:${INFOPATH}}
# ENV PCP_DIR=/opt/rh/devtoolset-7/root
# ENV PERL5LIB=/opt/rh/devtoolset-7/root//usr/lib64/perl5/vendor_perl:/opt/rh/devtoolset-7/root/usr/lib/perl5:/opt/rh/devtoolset-7/root//usr/share/perl5/
# ENV LD_LIBRARY_PATH=/opt/rocm/lib:/usr/local/lib:/opt/rh/devtoolset-7/root$rpmlibdir$rpmlibdir32${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
# ENV PYTHONPATH=/opt/rh/devtoolset-7/root/usr/lib64/python$pythonvers/site-packages:/opt/rh/devtoolset-7/root/usr/lib/python$pythonvers/
# ENV LDFLAGS="-Wl,-rpath=/opt/rh/devtoolset-7/root/usr/lib64 -Wl,-rpath=/opt/rh/devtoolset-7/root/usr/lib"
```

lets comment these env var as we are not using devtoolset7

File: amd/c9s-python-3.9/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+# RUN yum install -y devtoolset-7
+# RUN yum install -y devtoolset-7-libatomic-devel devtoolset-7-elfutils-libelf-devel
+
+# Install the ROCm rpms
+RUN yum clean all
+RUN echo -e "[ROCm]\nname=ROCm\nbaseurl=https://repo.radeon.com/rocm/yum/$ROCM_VERSION/main\nenabled=1\ngpgcheck=0" >> /etc/yum.repos.d/rocm.repo
+RUN echo -e "[amdgpu]\nname=amdgpu\nbaseurl=https://repo.radeon.com/amdgpu/$AMDGPU_VERSION/rhel/7.9/main/x86_64\nenabled=1\ngpgcheck=0" >> /etc/yum.repos.d/amdgpu.repo
+
+# unable to install 
+# RUN yum install -y rocm-dev rocm-libs
```
Comment (harshad16) [inline amd/c9s-python-3.9/Dockerfile:104]:
```suggestion
# Install rocm and amdgpu binaries 
# RUN dnf install -y amdgpu-dkms rocm
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
Thank for the great work,
provide few suggestion, to install amdgpu binaries.
and another request, can i have exactly same for ubi9 as well

**Comment by atheo89** [comment]
Great! Could you also address the `code static analysis` failures?

**Comment by harshad16** [comment]
tested these with : https://quay.io/repository/harshad16/amd-base?tab=tags
Screenshot from 2024-05-20 15-30-08

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

Excellent work @dibryant  :100: :medal_sports:

**Pull request merged**